### PR TITLE
Add New User Management

### PR DESCRIPTION
**Pull request by thenamankumar**
Closes https://github.com/sourcegraph/sourcegraph/issues/39821.

## Test plan
- `sg start`
- http://localhost:3080/site-admin/users?feature-flag-key=user-management&feature-flag-value=true

### Before
image 

### After
image

### CODE REVIEW

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -6382,6 +6382,7 @@ enum SiteUserOrderBy {
 SiteUsersLastActivePeriod enumerates the ways to filter based on users' last active date
 """
 enum SiteUsersLastActivePeriod {
+    ALL
```
Comment (erzhtor) [inline cmd/frontend/graphqlbackend/schema.graphql:6385]: Why do we need to add this? When this `nil`, it just doesn't filter by last active and thus behaves like all time anyway.

File: client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    selectable?: boolean
+    note?: string | JSX.Element
+    getRowId?: GetRowId<TData>
+    initialSortColumn?: string
+    initialSortDirection?: 'asc' | 'desc'
+    onSortChange?: (column: string | null, direction: 'asc' | 'desc' | null) => void
+    onSelectionChange?: (rows: TData[]) => void
+}
+
+type ISelectionState = { [key: string | number]: boolean }
```
Comment (erzhtor) [inline client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx:56]: This doesn't seem to be used anywhere?

File: client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+
+export default function Table<TData>({
```
Comment (erzhtor) [inline client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx:58]: Do you think we can move this to wildcard to make it reusable?

File: client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    selectable = false,
+    actions = [],
+    note,
+    getRowId = (data: any) => data.id,
+    onSortChange,
+    initialSortColumn = null,
+    initialSortDirection = 'asc',
+}: IProps<TData>) {
+    const [sortedColumn, setSortedColumn] = useState<string | null>(initialSortColumn)
+    const [sortDirection, setSortDirection] = useState<'asc' | 'desc' | null>(initialSortDirection)
```
Comment (erzhtor) [inline client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx:70]: I think this simply should be `boolean` type, smth like `isSortDescending = false`.

**Review by erzhtor — COMMENTED** [review]
Good job, looks great! I've left a few comments. P.S: It also seems there are quite a few linting errors (types, import orders, aria attributes, etc.).

File: client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx (hunk 3 above)
Comment (thenamankumar) [reply client/web/src/site-admin/SiteAdminAllUsersPage/New/components/Table/index.tsx]: I would like to first evolve and be more standardised and then collaborate with the frontend platform to make it more generic and then move it there.

File: cmd/frontend/graphqlbackend/schema.graphql (hunk 1 above)
Comment (thenamankumar) [reply cmd/frontend/graphqlbackend/schema.graphql:6385]: the horizontal selector component takes the label as value when the actual value is null for the option.

**Review by erzhtor — APPROVED** [review]

**Pull request merged**